src/java/org/apache/cassandra/config/DataRateSpec.java

```diff
@@ -228,6 +217,99 @@ public String toString()
+         *
+         * @param bytesPerSecond where bytesPerSecond shouldn't be bigger than Long.MAX_VALUE
+         */
+        public LongBytesPerSecondBound(long bytesPerSecond)
+        {
+            this(bytesPerSecond, BYTES_PER_SECOND);
+        }
+    }
+
+    /**
+     * Represents a data rate used for Cassandra configuration. The bound is [0; Integer.MAX_VALUE) in mebibytes per second.
+     * If the user sets a different unit - we still validate that converted to bytes per second the quantity will not exceed
+     * that upper bound. (CASSANDRA-17571)
+     */
+    public final static class IntMebibytesPerSecondBound extends DataRateSpec
+    {
+        /**
+         * Creates a {@code DataRateSpec.IntMebibytesPerSecondBound} of the specified amount with bound [0; Integer.MAX_VALUE) mebibytes per second.
+         *
+         * @param value the data rate
+         */
+        public IntMebibytesPerSecondBound(String value)
+        {
+            super(value, MEBIBYTES_PER_SECOND, Integer.MAX_VALUE);
+        }
+
+        /**
+         * Creates a {@code DataRateSpec.IntMebibytesPerSecondBound} of the specified amount in the specified unit.
+         *
+         * @param quantity where quantity shouldn't be bigger than Integer.MAX_VALUE - 1 in mebibytes per second
+         * @param unit     in which the provided quantity is
+         */
+        public IntMebibytesPerSecondBound(double quantity, DataRateUnit unit)
+        {
+            super(quantity, unit, MEBIBYTES_PER_SECOND, Integer.MAX_VALUE);
+        }
+
+        // this one should be used only for backward compatibility for stream_throughput_outbound and inter_dc_stream_throughput_outbound
+        // which were in megabits per second in 4.0. Do not start using it for any new properties
+        public static IntMebibytesPerSecondBound megabitsPerSecondInMebibytesPerSecond(long megabitsPerSecond)
+        {
+            final double MEBIBYTES_PER_MEGABIT = 0.119209289550781;
+            double mebibytesPerSecond = (double) megabitsPerSecond * MEBIBYTES_PER_MEGABIT;
+
+            if (megabitsPerSecond > Integer.MAX_VALUE)
+                throw new ConfigurationException("Invalid data rate: " + megabitsPerSecond + " megabits per second; " +
+                                                 "stream_throughput_outbound and inter_dc_stream_throughput_outbound" +
+                                                 " should be between 0 and " + Integer.MAX_VALUE + " in megabits per second");
+
+            return new IntMebibytesPerSecondBound(mebibytesPerSecond, MEBIBYTES_PER_SECOND);
+        }
+
+        /**
+         * Creates a {@code DataRateSpec.IntMebibytesPerSecondBound} of the specified amount in mebibytes per second.
+         *
+         * @param mebibytesPerSecond where mebibytesPerSecond shouldn't be bigger than Long.MAX_VALUE-1
+         */
+        public IntMebibytesPerSecondBound(long mebibytesPerSecond)
```

Comment:
nit: I think there are a few places around the code where we could use this version instead of the one that accepts a `String`, but perhaps that just because we want to test the value/unit parsing.

Reply:
I can see the String version in the test classes being tested and in Config. I guess we can use this one in the Config class, just feels more explicit there to stick to using the constructor version that is used by the yaml file. Also, we do it for all properties there, so we are consistent. (Except if someone else than me used some other constructor :D )

Reply:
I guess I can resolve this one?